src/components/NcButton/NcButton.vue

```diff
@@ -415,308 +430,314 @@ td.row-size {
+	 *
+	 * Note: This takes precedence over the href attribute.
+	 */
+	to?: string|RouteLocation
+}
 
-	computed: {
-		/**
-		 * The real type to be used for the button, enforces `primary` for pressed state and, if stateful button, any other type for not pressed state
-		 * Otherwise the type property is used.
-		 */
-		realType() {
-			// Force *primary* when pressed
-			if (this.pressed) {
-				return 'primary'
-			}
-			// If not pressed but button is configured as stateful button then the type must not be primary
-			if (this.pressed === false && this.type === 'primary') {
-				return 'secondary'
-			}
-			return this.type
-		},
+const props = withDefaults(defineProps<NcButtonProps>(), {
+	ariaLabel: undefined,
+	alignment: 'center',
+	download: undefined,
+	href: undefined,
+	pressed: undefined,
+	size: 'normal',
+	target: '_self',
+	text: undefined,
+	to: undefined,
+	type: 'button',
+	variant: 'secondary',
+})
 
-		/**
-		 * The flexbox alignment of the button content
-		 */
-		flexAlignment() {
-			return this.alignment.split('-')[0]
-		},
+const emit = defineEmits<{
+	/**
+	 * Emitted when the button was clicked.
+	 */
+	 click: [e: MouseEvent]
 
-		/**
-		 * If the button content should be reversed (icon on the end)
-		 */
-		isReverseAligned() {
-			return this.alignment.includes('-')
-		},
+	/**
+	 * Update the current pressed state of the button (if the `pressed` property was configured).
+	 * If the button is a toggle button (`pressed` state is boolean),
+	 * then this will be emitted if the user toggled the state.
+	 */
+	'update:pressed': [pressed: boolean]
+}>()
 
-		ncPopoverTriggerAttrs() {
-			return (this.getNcPopoverTriggerAttrs as () => Record<string, string|undefined>)()
-		},
-	},
+const slots = defineSlots<{
+	/**
+	 * Custom button content.
+	 * This can be used for custom formatted content, ensure to not include any interactive elements.
+	 * For plain text it is preferred to use the `text` prop instead.
+	 */
+	default?: Slot
 
 	/**
-	 * The render function to display the component
-	 *
-	 * @return {object|undefined} The created VNode
+	 * Icon (optional) to show within the button
 	 */
-	render() {
-		const hasText = isSlotPopulated(this.$slots.default?.())
-		const hasIcon = isSlotPopulated(this.$slots.icon?.())
-
-		/**
-		 * Always fill either the text prop or the ariaLabel one.
-		 */
-		if (!hasText && !this.ariaLabel) {
-			console.warn('You need to fill either the text or the ariaLabel props in the button component.', {
-				text: this.$slots.default?.()?.[0]?.children,
-				ariaLabel: this.ariaLabel,
-			},
-			this)
-		}
+	icon?: Slot
+}>()
+
+// Make sure the component also works if the app does not use any router
+// And if the app uses a router we need to make sure a `to` prop was passed to use to router
+const routerLink = computed(() => (
+	inject(routerKey, null) !== null && props.to
+		? useLink({ to: toRef(() => props.to as Required<typeof props.to>) })
```

Comment:
Isn't it the same as
```suggestion
		? useLink({ to: toRef(() => props.to!) })
```
?